### PR TITLE
PCHR-2610: Update T&A menu structure

### PR DESCRIPTION
**Pull request by AkA84**
## Overview
This PR updates the menu structure of the T&A extension

## Before
before-menu
before-menu-admin

## After
after-menu
after-menu-admin

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php [hunk 1]
```diff
@@ -591,12 +591,96 @@ public function upgrade_1026() {
...
-    {
-        CRM_Core_DAO::executeQuery("DELETE FROM `civicrm_navigation` WHERE name IN ('tasksassignments', 'ta_dashboard', 'tasksassignments_administer', 'ta_settings')");
-        CRM_Core_BAO_Navigation::resetNavigation();
+  /**
+   * Remove the old Dashboard submenu item
+   */
+  public function upgrade_1027() {
+    $taDashboard = new CRM_Core_DAO_Navigation();
+    $taDashboard->name = 'ta_dashboard';
+    $taDashboard->find(true);
```
Comment (mickadoo) [inline uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php]: Just to make the upgraders idempotent do you think it makes sense to put the same

```
if (!$taDashboard->id) {
  return TRUE;
}
```

check as below in this one?

File: uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php [hunk 2]
```diff
@@ -591,12 +591,96 @@ public function upgrade_1026() {
...
+  public function upgrade_1028() {
+    $taNavigation = new CRM_Core_DAO_Navigation();
+    $taNavigation->name = 'tasksassignments';
+    $taNavigation->find(true);
+    
+    if (!$taNavigation->id) {
+      return TRUE;
+    }
+
+    $submenu = array(
```
Comment (mickadoo) [inline uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php]: Array syntax

File: uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php [hunk 3]
```diff
@@ -591,12 +591,96 @@ public function upgrade_1026() {
...
+    return TRUE;
+  }
+
+  /**
+   * Add new submenu links
+   */
+  public function upgrade_1028() {
+    $taNavigation = new CRM_Core_DAO_Navigation();
+    $taNavigation->name = 'tasksassignments';
+    $taNavigation->find(true);
```
Comment (mickadoo) [inline uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php]: Mix of `TRUE` and `true`

File: uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php (hunk 1 above)
Comment (mickadoo) [inline uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php]: Mix of `TRUE` and `true`

File: uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php [hunk 4]
```diff
@@ -591,12 +591,96 @@ public function upgrade_1026() {
...
+    $paramsDashboard = ['name' => 'tasksassignments'];
+    $paramsAdmin = ['name' => 'tasksassignments_administer'];
+
+    $menuItems = [
+      'dashboard' => CRM_Core_BAO_Navigation::retrieve($paramsDashboard, $default),
+      'admin' => CRM_Core_BAO_Navigation::retrieve($paramsAdmin, $default)
+    ];
+    
+    foreach ($menuItems as $menuItem) {
+      $menuItem->label = 'Tasks';
```
Comment (mickadoo) [inline uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php]: Is it ever possible the `$menuItem` could be null?

Comment (AkA84) [reply uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php]: i would say it's impossible

File: uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php [hunk 5]
```diff
@@ -591,12 +591,96 @@ public function upgrade_1026() {
...
-        CRM_Core_DAO::executeQuery("DELETE FROM `civicrm_navigation` WHERE name IN ('tasksassignments', 'ta_dashboard', 'tasksassignments_administer', 'ta_settings')");
-        CRM_Core_BAO_Navigation::resetNavigation();
+  /**
+   * Remove the old Dashboard submenu item
+   */
+  public function upgrade_1027() {
+    $taDashboard = new CRM_Core_DAO_Navigation();
+    $taDashboard->name = 'ta_dashboard';
+    $taDashboard->find(TRUE);
+
```
Comment (mickadoo) [inline uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php]: Just to make the upgraders idempotent do you think it makes sense to put the same

```
if (!$taDashboard->id) {
  return TRUE;
}
```

check as below in this one?

File: uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php [hunk 6]
```diff
@@ -591,12 +591,96 @@ public function upgrade_1026() {
...
+    ];
+    
+    foreach ($menuItems as $menuItem) {
+      $menuItem->label = 'Tasks';
+      $menuItem->save();
+    }
+
+    CRM_Core_BAO_Navigation::resetNavigation();
+
+    return true;
```
Comment (mickadoo) [inline uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php]: One more :smile:

**Pull request merged**